src/platform/nrfconnect/OTAImageProcessorImpl.cpp

```diff
@@ -45,23 +46,12 @@ CHIP_ERROR OTAImageProcessorImpl::PrepareDownloadImpl()
     ReturnErrorOnFailure(System::MapErrorZephyr(dfu_target_mcuboot_set_buf(mBuffer, sizeof(mBuffer))));
     ReturnErrorOnFailure(System::MapErrorZephyr(dfu_target_reset()));
 
-    return System::MapErrorZephyr(dfu_target_init(DFU_TARGET_IMAGE_TYPE_MCUBOOT, /* size */ 0, nullptr));
+    // Initialize dfu target to receive first image
+    return System::MapErrorZephyr(dfu_target_init(DFU_TARGET_IMAGE_TYPE_MCUBOOT, mCurrentImage, /* size */ 0, nullptr));
```

Comment:
I think the first init should be called in `ProcessHeader` based on the image ID stored in the header. We should at least vierify that the first image is the app core image. `mCurrentImage` only store an image index in the OTA package, not the DFU target image ID.